### ISSUE TITLE
Support for equivalent mnemonics

### ISSUE THREAD

**Issue by ceharris** [gorsat/6809#2]
Most 6809 assemblers accept LSL as a synonym for ASL. This can be handled with a trivial macro, but it would be cool if it was recognized by the assembler.

**Comment by gorsat** [comment]
I added LSL. That seems to be an official mnemonic even though it's just an alias for ASL. Not sure why I left it out in the first place! 

I did a quick search to see if there were others like this but didn't find any. I see that some folks added convenience mnemonics to their assemblers (like SEC for "set carry") but didn't see any other "official" mnemonics. I'm not opposed to adding some convenience mnemonics if they're really useful (like, say, for porting a bunch of 6800 code to 6809 or something).

**Comment by ceharris** [comment]
I had that same thought yesterday -- one could pretty easily implement all of the 6800 compatibility mnemonics with relatively trivial macros (example `DEX -> LEAX -1,X`), so they'd probably be easy to recognize and substitute the corresponding 6809 opcode sequence. Maybe fun as an exercise, but I doubt it'd see much use. I suspect that most folks who are interested in this work probably aren't coming from a different 6800 series processor.

One other fairly common pseudo-op implemented in 6809 assemblers is SETDP, to inform the assembler of which memory accesses should implicitly use direct-page addressing. I didn't see it in your docs, but I admit I haven't had a chance to see if you implemented it -- I'm on a company machine at the moment, so I can't just do a quick check.  😁

**Comment by gorsat** [comment]
This led to 2 more issues: 

The code supports forcing direct mode with syntax like ``` LDA <TABLE ``` and forcing extended mode with syntax like ``` LDA >TABLE ```. However, I found that the code would reject forced direct mode when the supplied address was beyond zero page (>0xff). That's issue #3 which I believe is now fixed. 

I was looking at this to see how difficult it would be to add SETDP -- which seems like a really useful assembler directive -- and noticed that the current code has some questionable logic to determine whether DP has been touched or not and, in its current form, it will only try to automatically use direct mode if it thinks DP hasn't been touched and if the address in question is in the zero page. So I'm going to have to go through it a bit more and read some old reference material before implementing SETDP. This is now issue #4.
